src/crypto/secretbox/error.rs

```diff
@@ -0,0 +1,25 @@
+//! Error types for the `secretbox` module
+
+/// A type representing error conditions of secret box operations
+#[derive(Debug, PartialEq)]
+pub enum SecretBoxError {
+    /// The ciphertext was too short. This results from using the `crypto_secretbox_open_easy`
+    /// function, which expects a message prepended with a MAC.
+    CiphertextTooShort,
+    /// The ciphertext failed verification.
+    CiphertextFailedVerification,
+}
+
+#[cfg(feature = "std")]
+impl std::fmt::Display for SecretBoxError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self)
```

Comment:
Normally, the `Display` trait is used to provide human-readable error messages, target at end users. Won't it make better sense to display what is currently written in doc comments instead?

Reply:
Agreed. Do you know of an easy way to do this wthat doesn't involve duplicating the text in the doc comments?

Reply:
The only way I'm aware of is to use a custom macro :) like that one: [displaydoc](https://crates.io/crates/displaydoc) crate